code/modules/hydroponics/beekeeping/beebox.dm

```diff
@@ -7,23 +7,32 @@
 
 
 
-/mob/proc/bee_friendly()
-	return 0
+/mob/proc/bee_friendly() //Because we, are, your friends
+	return FALSE //You'll, never be(e) alone again    //NB, returning 0 here means that the bees are actually not your friends (they lied to you)
 
+/mob/living/simple_animal/hostile/poison/bees/bee_friendly() 
+	return TRUE //bee together strong
 
-/mob/living/simple_animal/hostile/poison/bees/bee_friendly()
-	return 1
+/mob/living/bee_friendly()
+	if(HAS_TRAIT(src, TRAIT_PIERCEIMMUNE)) //send your swarms but they'll never be(e) enough
+		return TRUE //my shell's too tough
+	return ..() //Maui man
 
+/mob/living/carbon/alien/humanoid/royal/bee_friendly()
+	return TRUE //can_inject() returns false for royal xenomorphs, so I guess they should no-sell bees too
+
+/mob/living/silicon/bee_friendly() //I wanted to do the Shiny joke here
+	return TRUE //but then I wouldn't have had enough space left for "Maui man"
```

Comment:
I think instead of all these that just return TRUE, make it a trait instead, and have the base of `bee_friendly` check for that trait.

Reply:
but then how would bees checking for thick clothing work?

Reply:
You'd still have the proc for anything with custom behavior, but it'd get rid of all the TRUE.